test/backend_pool.in.cpp

```diff
@@ -1235,3 +1235,84 @@ NGRAPH_TEST_P(${BACKEND_NAME}, avg_pool_3d_params, avg_pool_3d_uneven_strided_pa
 
 // avg_pool_3d case generation
 NGRAPH_INSTANTIATE_TEST_CASE_P(${BACKEND_NAME}, include_pad, avg_pool_3d_params, testing::Bool());
+
+//
+// This test primarly checks that maxpool backprop functions
+// correctly when the input tensor is larger than most cache sizes.
+// Here the to-be-pooled tensor is rank 2 with one non-trivial
+// dimension:
+//
+// x : [[0, 1, 0, 1, 0, 1, ... , 0, 1]]  <--- input data
+//       ----  ----  ----  ...   ----    <--- pooling windows
+// y : [[ 1  ,  1  ,  1  , ... ,  1]]    <--- max pooled output
+//
+// The pooling window is size 2 and stride 2, so the windows
+// do not overlap. Thus, each window will effectively see [0, 1]
+// as its input data for max pooling. The resulting output tensor
+// of pooling will be sizeof(x) with all elements equal to 1 as
+// seen above.
+// Therefore, for the backward pooling operation with the same window shape
+// and strides, the value of dy will only propogate to the positions in
+// dx that correspond to a value of 1 in the corresponding input tensor x:
+//
+// dy : [[2, 3, ... , 4]]
+// x  : [[0, 1, 0, 1, ... , 0, 1]]
+// dx : [[0, 2, 0, 3, ... , 0, 4]]
+//
+NGRAPH_TEST(${BACKEND_NAME}, maxpool_bprop_larger_than_cache)
+{
+    Shape window_shape{1, 2};
+    Strides move_strides{1, 2};
+    Shape padding_below{0, 0};
+    Shape padding_above{0, 0};
+
+    // 200 MB tensor to exceed cache
+    const size_t num_elements = 50 * 1024 * 1024;
```

Comment:
I increased the tensor size well past the identified failure condition so that if the failure is related to cache sizes, the test won't accidentally pass on future hardware (with larger caches).

Reply:
@shssf I had to disable this test on INTELGPU because the above chosen tensor size is too large. For the Titan XP used in CI, it needs to be at leas 3*1024 + 1 elements to spill into global DRAM. Ideally the number of elements chosen here is larger than any future GPU cache sizes. If you have a suggestion for something that meets this need but also can run on INTELGPU, I will update and re-enable the test.